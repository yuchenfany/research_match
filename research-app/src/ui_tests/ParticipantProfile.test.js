/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import ParticipantProfile from '../components/ParticipantProfile';

// const mockedUsedNavigate = jest.fn();

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useNavigate: () => mockedUsedNavigate,
// }));

test('Title Create User Profile', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantProfile user={user1} setUser={mockSetUser} />
        </Router>);  
  const label = screen.getByText("Create User Profile");
  expect(label).toBeInTheDocument();
});

test('Title Age', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantProfile user={user1} setUser={mockSetUser} />
        </Router>);  
  const label = screen.getByText("Medications");
  expect(label).toBeInTheDocument();
});


// test('find name text', () => {
//   const mockSetUser = jest.fn();
//   const user1 = { username: 'user1', password: 'user1pass' };
//   const mockSetStatus = jest.fn();
//   const status = false;
//   const mockSetStudy = jest.fn();
//   const study = {
//     studyId: '',
//     participants: [''],
//     researchers: [],
//     tags: [],
//   };
//     render(
//         <Router>
//             <ParticipantProfile user={user1} setUser={mockSetUser} />
//         </Router>);  
//   const label = screen.getByText("Name");
//   expect(label).toBeInTheDocument();
// });

// test('find create text', () => {
//   const mockSetUser = jest.fn();
//   const user1 = { username: 'user1', password: 'user1pass' };
//   const mockSetStatus = jest.fn();
//   const status = false;
//   const mockSetStudy = jest.fn();
//   const study = {
//     studyId: '',
//     participants: [''],
//     researchers: [],
//     tags: [],
//   };
//     render(
//         <Router>
//             <ParticipantProfile user={user1} setUser={mockSetUser} />
//         </Router>);  
//   const label = screen.getByText("Create Researcher Profile");
//   expect(label).toBeInTheDocument();
// });

// // test('participant clicked',() => {
// //   render(
// //     <Router>
// //         <Study />
// //     </Router>); 
// // // fireEvent.click(screen.getByTestId('participant'));
// // expect(screen.getByTestId('participant')).toBeInTheDocument();
// // //   expect(screen.getByTestId('participant')).toBeChecked();
// // });
test('Sign Up Button',() => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: ['all'],
    phys: ['phys'],
    psych: ['psych'],
    med: [],
    type: 0,
  };    
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantProfile user={user1} setUser={mockSetUser} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /sign up/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});


test('Sign Up Button',() => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };    
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantProfile user={user1} setUser={mockSetUser} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /sign up/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});

test('Sign Up Error Test Cases',() => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: '',
    password: '',
    enrolled: '',
    age: '',
    heightFeet: '',
    heightInches: '',
    weight: '',
    sex: '',
    gender: undefined,
    allergies: undefined,
    phys: undefined,
    psych: undefined,
    med: undefined,
    type: 0,
  };    
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantProfile user={user1} setUser={mockSetUser} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /sign up/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});

test('Sign Up Error Test Cases part2',() => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: '',
    password: '',
    enrolled: '',
    age: '',
    heightFeet: '',
    heightInches: '',
    weight: '',
    sex: undefined,
    gender: undefined,
    allergies: undefined,
    phys: undefined,
    psych: undefined,
    med: undefined,
    type: 0,
  };    
  const mockSetStatus = jest.fn();
  const status = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
    render(
        <Router>
            <ParticipantProfile user={user1} setUser={mockSetUser} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /sign up/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});

test('single input value  sequence', () => {
  const mockUser = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };  
  const mockSetUser = jest.fn();

  render(
    <Router>
      <ParticipantProfile user={mockUser} setUser={mockSetUser} />
    </Router>,
  );

  const input = screen.getByPlaceholderText('age');
  fireEvent.change(input, { target: { value: '10' } });
  expect(input.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  const input2 = screen.getByPlaceholderText('height');
  fireEvent.change(input2, { target: { value: '10' } });
  expect(input2.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  const input3 = screen.getByPlaceholderText('ft');
  fireEvent.change(input3, { target: { value: '10' } });
  expect(input3.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  const input4 = screen.getByPlaceholderText('weight');
  fireEvent.change(input4, { target: { value: '10' } });
  expect(input4.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  // fireEvent.click(screen.getByDisplayValue('SUBMIT'));
});

test('snapshot test', () => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  const component = renderer.create( 
    <Router>
        <ParticipantProfile user={user1} setUser={mockSetUser} />
    </Router>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//TODO: How to do textbox field testing for those functions