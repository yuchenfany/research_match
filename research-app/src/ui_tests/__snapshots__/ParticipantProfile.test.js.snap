// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Profile"
>
  <div
    className="profile-flex"
  >
    <div
      className="header-left"
    >
       Create User Profile 
    </div>
    <div
      className="profile-row"
    >
      <div>
        Age
      </div>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="age"
        type="text"
      />
      <span
        className="error-message"
      >
        
      </span>
      <div>
        Height
      </div>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="height"
        type="text"
      />
      <div>
        ft
      </div>
      <span
        className="error-message"
      >
        
      </span>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="ft"
        type="text"
      />
      <div>
        in
      </div>
      <span
        className="error-message"
      >
        
      </span>
      <div>
        Weight
      </div>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="weight"
        type="text"
      />
      <div>
        lbs
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Biological Sex
      </div>
      <label
        className="radio-option"
        htmlFor="form"
      >
        <input
          id="male"
          name="option"
          onClick={[Function]}
          type="radio"
          value="male"
        />
        <div>
          Male
        </div>
      </label>
      <label
        className="radio-option"
        htmlFor="form"
      >
        <input
          id="female"
          name="option"
          onClick={[Function]}
          type="radio"
          value="female"
        />
        <div>
          Female
        </div>
      </label>
      <label
        className="radio-option"
        htmlFor="form"
      >
        <input
          id="intersex"
          name="option"
          onClick={[Function]}
          type="radio"
          value="intersex"
        />
        <div>
          Intersex
        </div>
      </label>
    </div>
    <div
      className="profile-row"
    >
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Gender
      </div>
      <div
        className="dropdown"
      >
        <div
          className=" css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-37-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-108aapi-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-14el2xx-placeholder"
                id="react-select-37-placeholder"
              >
                Select...
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-37-listbox"
                  aria-describedby="react-select-37-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-owns="react-select-37-listbox"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-37-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Allergies
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className=" css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-38-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-38-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-38-listbox"
                aria-describedby="react-select-38-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-38-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-38-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Medical Conditions (Physical)
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className="select-tags css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-39-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-39-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-39-listbox"
                aria-describedby="react-select-39-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-39-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-39-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Medical Conditions (Psychological)
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className="select-tags css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-40-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-40-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-40-listbox"
                aria-describedby="react-select-40-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-40-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-40-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Medications
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className="select-tags css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-41-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-41-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-41-listbox"
                aria-describedby="react-select-41-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-41-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-41-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <input
      className="signup-button"
      onClick={[Function]}
      type="submit"
      value="SIGN UP"
    />
  </div>
</div>
`;
